rules/TypeDeclaration/Rector/ClassMethod/ArrayShapeFromConstantArrayReturnRector.php

```diff
@@ -168,4 +174,23 @@ private function shouldSkip(ConstantArrayType $constantArrayType): bool
 
         return false;
     }
+
+    /**
+     * Skip test case, as return methods there are usually with test data only.
+     * Those arrays are hand made and return types are getting complex and messy, so this rule should skip it.
+     */
+    private function isInTestCase(ClassMethod $classMethod): bool
+    {
+        $scope = $classMethod->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return false;
+        }
+
+        $classReflection = $scope->getClassReflection();
```

Comment:
there is `ReflectionResolver::resolveClassReflection()` for it.

Reply:
I'd go with this as it simple and available in both PHPStan and Rector natively. Both are fine :)